src/Analyzers/VisualBasic/CodeFixes/SimplifyObjectCreation/VisualBasicSimplifyObjectCreationCodeFixProvider.vb

```diff
@@ -0,0 +1,61 @@
+﻿' Licensed to the .NET Foundation under one or more agreements.
+' The .NET Foundation licenses this file to you under the MIT license.
+' See the LICENSE file in the project root for more information.
+
+Imports System.Collections.Immutable
+Imports System.Composition
+Imports System.Threading
+Imports Microsoft.CodeAnalysis.CodeActions
+Imports Microsoft.CodeAnalysis.CodeFixes
+Imports Microsoft.CodeAnalysis.Diagnostics
+Imports Microsoft.CodeAnalysis.Editing
+Imports Microsoft.CodeAnalysis.Host.Mef
+Imports Microsoft.CodeAnalysis.VisualBasic.Syntax
+
+Namespace Microsoft.CodeAnalysis.VisualBasic.SimplifyObjectCreation
+    <ExportCodeFixProvider(LanguageNames.VisualBasic), [Shared]>
+    Friend Class VisualBasicSimplifyObjectCreationCodeFixProvider
+        Inherits SyntaxEditorBasedCodeFixProvider
+
+        <ImportingConstructor>
+        <Obsolete(MefConstruction.ImportingConstructorMessage, True)>
+        Public Sub New()
+        End Sub
+
+        Public Overrides ReadOnly Property FixableDiagnosticIds As ImmutableArray(Of String) = ImmutableArray.Create(IDEDiagnosticIds.SimplifyObjectCreationDiagnosticId)
+
+        Friend Overrides ReadOnly Property CodeFixCategory As CodeFixCategory = CodeFixCategory.CodeStyle
+
+        Public Overrides Function RegisterCodeFixesAsync(context As CodeFixContext) As Task
+            For Each diagnostic In context.Diagnostics
+                context.RegisterCodeFix(New MyCodeAction(
+                    VisualBasicCodeFixesResources.Simplify_object_creation,
+                    Function(ct) FixAsync(context.Document, diagnostic, ct)),
+                    diagnostic)
+            Next
+            Return Task.CompletedTask
+        End Function
+
+        Protected Overrides Async Function FixAllAsync(document As Document, diagnostics As ImmutableArray(Of Diagnostic), editor As SyntaxEditor, cancellationToken As CancellationToken) As Task
+            Dim root = Await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(False)
+            For Each diagnostic In diagnostics
+                Dim node = DirectCast(root.FindNode(diagnostic.Location.SourceSpan, getInnermostNodeForTie:=True), VariableDeclaratorSyntax)
+                Dim asNewClause = SyntaxFactory.AsNewClause(node.AsClause.AsKeyword, DirectCast(node.Initializer.Value, NewExpressionSyntax))
```

Comment:
In the analyzer you check for Object creation. Here you cast to New expression. (I'm not at computer), is this safe?

Reply:
@CyrusNajmabadi Nice catch! I think the cast may fail. Will come up with a test and fix that.

Reply:
The cast doesn't fail since an `ObjectCreationExpressionSyntax` derives from `NewExpressionSyntax` (i.e, it's always safe to cast an `ObjectCreationExpressionSyntax` to its base type. However, the analyzer is missing cases.

Reply:
@CyrusNajmabadi The syntactic simplification of an expression like `Dim result As C() = New C() { }` (i.e, `ArrayCreationExpression`) will be different since `Dim result As New C() { }` isn't allowed by the language. Let me know if you want me to handle this or keep it simple and only catch `ObjectCreationExpressionSyntax`.

Reply:
just keep it simple.  make sure we have tests and whatnot.